packages/catalog-process/test/catalogService.integration.test.ts

```diff
@@ -2902,15 +2909,132 @@ describe("database test", async () => {
+      });
+      it("should throw eServiceNotFound if there are no descriptors (requester is not the producer)", async () => {
+        const eservice: EService = {
+          ...mockEService,
+          descriptors: [],
+        };
+        await addOneEService(mockEService, postgresDB, eservices);
+        expect(
+          catalogService.getEServiceById(eservice.id, getMockAuthData())
+        ).rejects.toThrowError(eServiceNotFound(eservice.id));
+      });
+      it("should throw eServiceNotFound if there are no descriptors (requester is the producer but not admin nor api)", async () => {
+        const descriptor = {
+          ...mockDescriptor,
+          state: descriptorState.draft,
+        };
+        const eservice: EService = {
+          ...mockEService,
+          descriptors: [descriptor],
+        };
+        const authData: AuthData = {
+          ...getMockAuthData(eservice.producerId),
+          userRoles: [userRoles.SUPPORT_ROLE],
+        };
+        await addOneEService(mockEService, postgresDB, eservices);
+        expect(
+          catalogService.getEServiceById(eservice.id, authData)
+        ).rejects.toThrowError(eServiceNotFound(eservice.id));
+      });
+      it("should filter out the draft descriptors if the eService has both draft and non-draft ones (requester is not the producer)", async () => {
+        const descriptorA: Descriptor = {
+          ...mockDescriptor,
+          state: descriptorState.draft,
+        };
+        const descriptorB: Descriptor = {
+          ...mockDescriptor,
+          state: descriptorState.published,
+          interface: mockDocument,
+          publishedAt: new Date(),
+        };
+        const eservice: EService = {
+          ...mockEService,
+          descriptors: [descriptorA, descriptorB],
+        };
+        const authData: AuthData = {
+          ...getMockAuthData(),
+          userRoles: [userRoles.ADMIN_ROLE],
+        };
+        await addOneEService(eservice, postgresDB, eservices);
+        const result = await catalogService.getEServiceById(
+          eservice.id,
+          authData
+        );
+        expect(result.descriptors).toEqual([descriptorB]);
+      });
+      it("should filter out the draft descriptors if the eService has both draft and non-draft ones (requester is the producer but not admin nor api)", async () => {
+        const descriptorA: Descriptor = {
+          ...mockDescriptor,
+          state: descriptorState.draft,
+        };
+        const descriptorB: Descriptor = {
+          ...mockDescriptor,
+          state: descriptorState.published,
+          interface: mockDocument,
+          publishedAt: new Date(),
+        };
+        const eservice: EService = {
+          ...mockEService,
+          descriptors: [descriptorA, descriptorB],
+        };
+        const authData: AuthData = {
+          ...getMockAuthData(eservice.producerId),
+          userRoles: [userRoles.SUPPORT_ROLE],
```

Comment:
@beetlecrunch just noticed that with the new requirement `SUPPORT` role will not be able to see drafts, is it ok?
(in Scala it is implemented the same behaviour)

Reply:
> @beetlecrunch just noticed that with the new requirement `SUPPORT` role will not be able to see drafts, is it ok? (in Scala it is implemented the same behaviour)

Me and @ruggerocastagnola noticed this too and we thought that for now it's ok but in the future would be better if `SUPPORT` could see drafts.